include/graybat/Cage.hpp

```diff
@@ -683,6 +692,29 @@ namespace graybat {
     template<typename T_CommunicationPolicy, typename T_GraphPolicy>
     template<typename T>
     auto Cage<T_CommunicationPolicy, T_GraphPolicy>::
+	recv (const Edge &edge, std::future<T>& data)
+	-> void
+	{
+		VAddr srcVAddr = locateVertex(edge.source);
+		std::promise<T> dataPromise;
+		data = dataPromise.get_future();
+
+		// The reason for the std::promise and std::thread over std::async ist the destructor behaviour of std::async
+		// For std::async the destructor is waiting for the completion of the asyncThread, which may lead to a permanent lock
+		// The future from std::promise does not block on the completion of the execution thread.
+		std::thread asyncRecvThread([&](){
+			T result;
+			Event e = comm->asyncRecv(srcVAddr, edge.id, graphContext, result);
+			e.wait();
+			dataPromise.set_value(result);
+		});
+		asyncRecvThread.detach();
+	}
```

Comment:
I don't understand, where you would want to go with such a policy and how to sensefully isolate such a policy.
Can you make a little draft, how you want it to be, than i can make the changes.